### PR TITLE
refactor: adapt "DatasetManager" with OpenAPIv2

### PR DESCRIPTION
**Pull request by graczhual**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2k67b5k [parent] 对接基础 OpenAPIv2 与 Manager](https://app.clickup.com/t/2k67b5k)

File: graviti/graviti.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 """The implementation of the Graviti."""
 
 from graviti.manager import DatasetManager
+from graviti.openapi import get_current_user
```
Comment (linjiX) [inline graviti/graviti.py:9]: The commit message is not clear

File: graviti/manager/dataset.py [hunk 2]
```diff
@@ -5,34 +5,30 @@
...
 from graviti.schema import Extractors, catalog_to_schema, get_extractors
-from graviti.utility import LazyFactory, LazyList, NestedDict
+from graviti.utility import LazyFactory, LazyList, NestedDict, ReprMixin
 
 LazyLists = NestedDict[str, LazyList[Any]]
 
 
-class Dataset(Mapping[str, DataFrame]):  # pylint: disable=too-many-instance-attributes
+class Dataset(  # pylint: disable=too-many-instance-attributes
+    AttrsMixin, ReprMixin, Mapping[str, DataFrame]
```
Comment (linjiX) [inline graviti/manager/dataset.py:30]: shall we set `_repr_type`?

File: graviti/manager/dataset.py [hunk 3]
```diff
@@ -41,12 +37,44 @@ class Dataset(Mapping[str, DataFrame]):  # pylint: disable=too-many-instance-att
...
+    _name: str = attr(key="name")
+    _alias: str = attr(key="alias")
+    _default_branch: str = attr(key="default_branch")
+    _commit_id: str = attr(key="commit_id")
+    created_at: str = attr()
+    updated_at: str = attr()
+    _owner: str = attr(key="owner")
+    _is_public: bool = attr(key="is_public")
+    _config: str = attr(key="config")
+    _branch: Optional[str] = attr(key="branch")
```
Comment (linjiX) [inline graviti/manager/dataset.py:77]: why branch is optional？

File: graviti/manager/dataset.py [hunk 4]
```diff
@@ -315,47 +412,56 @@ class DatasetManager:
...
+            The created :class:`~graviti.manager.dataset.Dataset` instance.
 
         """
+        response = create_dataset(
+            self._access_key,
+            self._url,
+            name,
+            alias=alias,
+            is_public=is_public,
+            config=config,
```
Comment (linjiX) [inline graviti/manager/dataset.py:454]: shall we put name, alias, is_public ... into a dict?

File: graviti/manager/dataset.py [hunk 5]
```diff
@@ -56,18 +84,28 @@ def __init__(
...
         alias: str,
+        default_branch: str,
         commit_id: str,
-        branch_name: Optional[str] = None,
+        created_at: str,
+        updated_at: str,
+        owner: str,
+        is_public: bool,
+        config: str,
+        branch: Optional[str] = None,
```
Comment (linjiX) [inline graviti/manager/dataset.py:94]: can we remove `branch` here just use `default_branch`?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/manager/dataset.py [hunk 6]
```diff
@@ -41,12 +37,44 @@ class Dataset(Mapping[str, DataFrame]):  # pylint: disable=too-many-instance-att
...
+    )
+
+    _access_key: str = attr(key="access_key")
+    _url: str = attr(key="url")
+    _dataset_id: str = attr(key="id")
+    _name: str = attr(key="name")
+    _alias: str = attr(key="alias")
+    _default_branch: str = attr(key="default_branch")
+    _commit_id: str = attr(key="commit_id")
+    created_at: str = attr()
```
Comment (linjiX) [inline graviti/manager/dataset.py:72]: why `created_at` is not property?

File: graviti/manager/dataset.py (hunk 3 above)
Comment (graczhual) [reply graviti/manager/dataset.py:77]: when we checkout to a commit, the branch will be None

File: graviti/manager/dataset.py (hunk 5 above)
Comment (graczhual) [reply graviti/manager/dataset.py:94]: branch is useful when we checkout to other branch

File: graviti/manager/dataset.py (hunk 6 above)
Comment (graczhual) [reply graviti/manager/dataset.py:72]: I'm not sure. I just don't think this attribute is important, and it doesn't matter if users change it. What should we use to determine when to use property? the attribute is not modifiable?

File: graviti/manager/dataset.py (hunk 3 above)
Comment (linjiX) [reply graviti/manager/dataset.py:77]: got it

File: graviti/manager/dataset.py (hunk 5 above)
Comment (linjiX) [reply graviti/manager/dataset.py:94]: can we just set `branch = default_branch` in `__init__` and remove the `branch` as input?

File: graviti/manager/dataset.py [hunk 7]
```diff
@@ -41,12 +37,44 @@ class Dataset(Mapping[str, DataFrame]):  # pylint: disable=too-many-instance-att
...
+        updated_at: The time of last update.
+        owner: The owner of the dataset.
+        is_public: Whether the dataset is public.
+        config: The config of dataset.
+        branch: The current branch name of the dataset.
 
     """
 
     _data: Dict[str, DataFrame]
+    _T = TypeVar("_T", bound="Dataset")
```
Comment (Lee-000) [inline graviti/manager/dataset.py:52]: Should the order of the variables follow the standard?

File: graviti/manager/dataset.py (hunk 2 above)
Comment (graczhual) [reply graviti/manager/dataset.py:30]: _repr_type = ReprType.INSTANCE
do not need to set here, this is the default value

Comment (linjiX) [reply graviti/manager/dataset.py:30]: yes the key is sheet name, will it make something wrong ?

Comment (graczhual) [reply graviti/manager/dataset.py:30]: So when we get a dataset, all segments will be initialized?

Comment (linjiX) [reply graviti/manager/dataset.py:30]: ReprMixin only effects `__repr__`

File: graviti/manager/dataset.py [hunk 8]
```diff
@@ -315,47 +432,48 @@ class DatasetManager:
...
-            yield Dataset(
-                self._access_key,
-                self._url,
-                item["id"],
-                item["name"],
-                branch_name=item["defaultBranch"],
-                commit_id=item["commitId"],
-                alias=item["alias"],
-            )
+            yield Dataset.from_pyobj(item)
```
Comment (Lee-000) [inline graviti/manager/dataset.py:444]: Does `item` contain `accesskey` and `url`?

File: graviti/manager/dataset.py [hunk 9]
```diff
@@ -315,47 +432,48 @@ class DatasetManager:
...
+        arguments: Dict[str, Any] = {
+            "name": name,
+            "alias": alias,
+            "is_public": is_public,
+            "config": config,
+        }
+        response = create_dataset(self._access_key, self._url, **arguments)
+        response.update(arguments)
+
+        return Dataset.from_pyobj(response)
```
Comment (Lee-000) [inline graviti/manager/dataset.py:476]: Why is it necessary to add the method `from_pyobj`? Would `init` do?

File: graviti/manager/dataset.py [hunk 10]
```diff
@@ -41,11 +37,44 @@ class Dataset(Mapping[str, DataFrame]):  # pylint: disable=too-many-instance-att
...
+        "created_at",
+        "updated_at",
+        "is_public",
+        "config",
+        "branch",
+    )
+
+    access_key: str = attr()
+    url: str = attr()
+    dataset_id: str = attr(key="id")
```
Comment (linjiX) [inline graviti/manager/dataset.py:67]: we can assume that the dataset id is useless for user, shall we make it private and remove it from the _repr_attrs

File: graviti/graviti.py (hunk 1 above)
Comment (linjiX) [reply graviti/graviti.py:9]: How about 'adapt "DatasetManager" with OpenAPIv2'

File: graviti/manager/dataset.py (hunk 9 above)
Comment (graczhual) [reply graviti/manager/dataset.py:347]: some response params are different from init

File: graviti/manager/dataset.py (hunk 10 above)
Comment (graczhual) [reply graviti/manager/dataset.py:67]: ok

**Review by linjiX — APPROVED** [review]

**Review by Lee-000 — APPROVED** [review]

**Pull request merged**